### PR TITLE
Convert MakeEdgeScal Fortran kernels to C++ lambdas

### PR DESCRIPTION
**Pull request by harpolea**
This converts the Fortran kernel launches in the `MakeEdgeScal` routine to C++ lambda functions. The performance on CPU is essentially unchanged, but on GPU I see a speed up of 20-30x for this function, leading to an speed up overall of ~30% for `Maestro::Evolve`. 

The actual code itself is essentially unchanged - I've just added punctuation and fixed the array indexing to match C++ counting from 0. 

I've tested this for 3d `reacting_bubble` and `wdconvect` and the answers seem to match development.

Conversion of the 2d routines and slope routines (for ppm type 0) will come in later PRs.

### CODE REVIEW

File: Source/MaestroHydro.cpp [hunk 1]
```diff
@@ -308,10 +312,12 @@ Maestro::MakeUtrans (const Vector<MultiFab>& utilde,
                         BL_TO_FORTRAN_ANYD(w_mf[mfi]),
                         BL_TO_FORTRAN_ANYD(Ip[mfi]),
                         BL_TO_FORTRAN_ANYD(Im[mfi]),
+                        BL_TO_FORTRAN_ANYD(u_mf[mfi]),
+                        BL_TO_FORTRAN_ANYD(v_mf[mfi]),
```
Comment (zingale) [inline Source/MaestroHydro.cpp]: this is confusing -- why `u_mf` and `v_mf` but not `w_mf`?  Are these arguments even needed?

Comment (harpolea) [reply Source/MaestroHydro.cpp]: oh these are dummy arguments I added when I was tested the ppm_3d implementation. I'll clean that up

File: Source/MaestroMakeEdgeScalars.cpp [hunk 2]
```diff
@@ -0,0 +1,1100 @@
...
+    Real dx_local = dx[0];
+
+    AMREX_PARALLEL_FOR_3D(bx, i, j, k, 
+    {
+        divu(i,j,k) = 
+            umac(i+1,j,k) - umac(i,j,k) +
+            vmac(i,j+1,k) - vmac(i,j,k) +
+            wmac(i,j,k+1) - wmac(i,j,k);
+        divu(i,j,k) /= dx_local;
+    });
```
Comment (zingale) [inline Source/MaestroMakeEdgeScalars.cpp]: not sure how it optimizes, but maybe precompute 1/dx_local outside the lambda and multiply

File: Source/MaestroMakeEdgeScalars.cpp [hunk 3]
```diff
@@ -0,0 +1,1100 @@
...
+    // loop over appropriate y-faces
+    int jlo = domainBox.loVect()[1];
+    int jhi = domainBox.hiVect()[1];
+    bclo = bcs[bccomp].lo()[1];
+    bchi = bcs[bccomp].hi()[1];
+    AMREX_PARALLEL_FOR_3D(mybx, i, j, k, 
+    {
+        if (ppm_type_local == 0) {
+            sly(i,j,k) = scal(i,j-1,k,comp) + 
+                0.5 * (1.0 - dt * vmac(i,j,k) / hy) * Ip(i,j-1,k,0);
```
Comment (zingale) [inline Source/MaestroMakeEdgeScalars.cpp]: should these slopes have the last index of `1` instead of `0`?

Comment (zingale) [reply Source/MaestroMakeEdgeScalars.cpp]: or be `Im`?

Comment (harpolea) [reply Source/MaestroMakeEdgeScalars.cpp]: no, but they should be Im not Ip. For ppm type 0, Ip stores slopex and Im stores slopey

File: Source/MaestroMakeEdgeScalars.cpp [hunk 4]
```diff
@@ -0,0 +1,1100 @@
...
+        Real srzy = 0.0;
+        // loop over appropriate zy faces
+        if (is_conservative) {
+            // make slzy, srzy by updating 1D extrapolation
+            slzy = slz(i,j,k) 
+                - (dt3/hy) * (simhy(i,j+1,k-1)*vmac(i,j+1,k-1) 
+                - simhy(i,j,k-1)*vmac(i,j,k-1)) 
+                - dt3*scal(i,j,k-1,comp)*divu(i,j,k-1) 
+                + (dt3/hy)*scal(i,j,k-1,comp)*
+                (umac(i,j+1,k-1)-vmac(i,j,k-1));
```
Comment (zingale) [inline Source/MaestroMakeEdgeScalars.cpp]: umac - vmac?
shouldn't this be vmac - vmac?

**Pull request merged**